File: app/src/main/java/pl/butajlo/androidadvanced/models/ZoneDateTimeAdapter.java
package pl.butajlo.androidadvanced.models;

import android.support.annotation.Nullable;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import org.threeten.bp.ZonedDateTime;

public class ZoneDateTimeAdapter {

    @FromJson
    ZonedDateTime fromJson(String json) {
        return ZonedDateTime.parse(json);
    }

    @ToJson
    String toJson(@Nullable ZonedDateTime zdt) {
        return zdt != null ? zdt.toString() : null;
    }

}

File: app/src/main/java/pl/butajlo/androidadvanced/models/ZoneDateTimeAdapter.kt
package pl.butajlo.androidadvanced.models

import android.support.annotation.Nullable
import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson
import org.threeten.bp.ZonedDateTime

class ZoneDateTimeAdapter {
    
    @FromJson
    fun fromJson(json: String): ZonedDateTime = ZonedDateTime.parse(json)
    
    @ToJson
    fun toJson(zdt: ZonedDateTime?) = zdt.toString()
}